client/my-sites/stats/activity-log-tasklist/index.jsx

```diff
@@ -0,0 +1,290 @@
+					} ),
+					{
+						showDismiss: false,
+					}
+				),
+			} ),
+		} );
+	};
+
+	componentWillReceiveProps( nextProps ) {
+		if ( isEmpty( nextProps.pluginsToUpdate ) ) {
+			return;
+		}
+
+		const {
+			removeThisNotice,
+			showErrorNotice,
+			showSuccessNotice,
+			siteName,
+			translate,
+			pluginsToUpdate,
+		} = nextProps;
+		const newState = {};
+		const noticesToShow = {};
+		const pluginsCompleted = [];
+
+		each( Object.values( pluginsToUpdate ), plugin => {
+			const pluginSlug = plugin.slug;
+
+			if ( false === get( this.props.pluginsToUpdate, `${ pluginSlug }.updateStatus`, false ) ) {
+				return;
+			}
+
+			if (
+				get( this.props.pluginsToUpdate, `${ pluginSlug }.updateStatus.status`, false ) ===
+					get( plugin.updateStatus, 'status', false ) ||
+				isPluginUpdating( plugin )
+			) {
+				return;
+			}
+
+			const updateStatus = plugin.updateStatus;
+			const updateNoticeId = get(
+				this.state.pluginUpdateNotice,
+				[ pluginSlug, 'notice', 'noticeId' ],
+				null
+			);
+
+			// If there is no notice displayed
+			if ( ! updateNoticeId ) {
+				return;
+			}
+
+			removeThisNotice( updateNoticeId );
+			noticesToShow[ pluginSlug ] = null;
+
+			// If it errored, clear and show error notice
+			const pluginData = {
+				plugin: plugin.name,
+				site: siteName,
+			};
+
+			switch ( updateStatus.status ) {
+				case 'error':
+					noticesToShow[ pluginSlug ] = showErrorNotice(
+						PluginNotices.singleErrorMessage( 'UPDATE_PLUGIN', pluginData, {
+							error: updateStatus,
+						} ),
+						{
+							button: translate( 'Try again' ),
+							onClick: () => this.updatePlugin( plugin ),
+						}
+					);
+					break;
+				case 'completed':
+					noticesToShow[ pluginSlug ] = showSuccessNotice(
+						PluginNotices.successMessage( 'UPDATE_PLUGIN', '1 site 1 plugin', pluginData ),
+						{
+							duration: 5555,
+						}
+					);
+					pluginsCompleted.push( plugin.slug );
+					break;
+			}
+		} );
+
+		if ( ! isEmpty( noticesToShow ) ) {
+			newState.pluginUpdateNotice = merge( {}, this.state.pluginUpdateNotice, noticesToShow );
+		}
+
+		if ( ! isEmpty( pluginsCompleted ) ) {
+			newState.dismissedPlugins = this.state.dismissedPlugins.concat( pluginsCompleted );
+		}
+
+		if ( ! isEmpty( newState ) ) {
+			this.setState( newState );
+		}
+	}
+
+	/**
+	 * Checks if the supplied plugin didn't start its update yet.
+	 * This is the default plugin status.
+	 *
+	 * @param {object} plugin Plugin to check status
+	 * @returns {bool} True if the update didn't started.
+	 */
+	isPluginUpdateNotStarted = plugin => false === plugin.updateStatus;
+
+	render() {
+		if ( this.state.dismissAll ) {
+			return null;
+		}
+
+		const { translate, pluginsWithUpdates } = this.props;
+		const dismissedPlugins = this.state.dismissedPlugins;
+		const pluginsToUpdate = Object.values( omit( this.props.pluginsToUpdate, dismissedPlugins ) );
+		const numberOfPluginUpdates = pluginsWithUpdates.length - dismissedPlugins.length;
```

Comment:
also presumably this is just `pluginsToUpdate.length`

Reply:
Not really. `pluginsToUpdate` includes all plugins, updated or pending. `pluginsWithUpdates` includes those that have pending updates, and `dismissedPlugins` should be subtracted since they might have been dismissed and not updated.